src/components/tools/Map/VectorLayer.svelte

```diff
@@ -52,10 +53,16 @@
 
   if (enableClick) {
     map.on("click", layer.id, function (e) {
-      dispatch("overlayclick", e.features[0].id);
-      map.flyTo({
-        center: e.features[0].geometry.coordinates,
-      });
+      const feature = e.features[0];
+      dispatch("overlayclick", feature.id);
+      let center;
+      if (feature.geometry === "Point") {
+        center = feature.geometry.coordinates;
+      } else {
+        const pointFeature = getCenter(feature.geometry);
```

Comment:
It might be good to extract the second half of that `else` statement into a separate function and wrap its contents in an `try / catch` statement in case the call to `getCenter` fails for some reason. I guess then you would need to put an `if` around `map.flyTo` to make sure it only fires if `center` is defined.